core/src/main/scala/kafka/coordinator/GroupMetadata.scala

```diff
@@ -264,16 +265,17 @@ private[coordinator] class GroupMetadata(val groupId: String, initialState: Grou
   }
 
   def completePendingOffsetWrite(topicPartition: TopicPartition, offset: OffsetAndMetadata) {
-    offsets.put(topicPartition, offset)
+    if(!deletedPartitions.contains(topicPartition))
+      offsets.put(topicPartition, offset)
     pendingOffsetCommits.get(topicPartition) match {
-      case Some(stagedOffset) if offset == stagedOffset => pendingOffsetCommits.remove(topicPartition)
+      case Some(stagedOffset) if offset == stagedOffset => pendingOffsetCommits.remove(topicPartition); deletedPartitions -= topicPartition
```

Comment:
style nit: probably no need to fit onto a single line. Can we just do this?
```scala
case Some(stagedOffset) if offset == stagedOffset =>
  pendingOffsetCommits.remove(topicPartition)
  deletedPartitions -= topicPartition
```
Similar below.